dashboard/app/controllers/aichat_controller.rb

```diff
@@ -34,6 +34,10 @@ def start_chat_completion
 
     # Create the request object
     begin
+      # JIRA LABS-1294: These `to_json`s on the json fields (model_customizations, stored_messages, and new_message)
+      # are json stringifying the ruby hashes (objects) before storing them, so they are stored as escaped strings.
+      # We should remove these and either handle both cases (parse the json if it's a string when we read) or do
+      # a one-time migration to parse all the strings and rewrite them.
```

Comment:
To confirm my understanding, we should remove `to_json` so we don't have to parse the json when accessing these values?

Reply:
Yes! The way we do it now, we are treating it as if the field in the DB is a string rather than taking advantage of the native json field.

Reply:
Logging here that we discussed this topic alongside @sanchitmalhotra126. We can add similar code to `AichatRequests` since chat logs are removed after 90 days and then eventually remove the the check for a string.

Also noting that https://codedotorg.atlassian.net/browse/LABS-1213 that should probably be picked up by someone else (happy to pick up since @sanchitmalhotra126 is going to work on follow-ups to this PR).